doc/memos/rdm3.md

```diff
@@ -0,0 +1,104 @@
+- RIOT Maintainers Community
+- Developer Memo: 3
+- Author(s): Martine Lenders
+- Date: January 2017
+
+***
+
+# Network programming guide
+
+## Abstract
+This memo describes and justifies the usage and existence of RIOT's various
+networking APIs.
+
+
+## Status
+This document is a product of the community of RIOT maintainers, and aims to 
+represent the consensus within this community.
+The content of this document is licensed with a Creative Commons CC-BY-SA 
+license.
+
+## Terminology
+This memo uses the following terminology and acronyms:
+
+- network stack: the architecture to implement networking
+- GNRC: RIOT's default network stack
+- netapi: GNRC's inter-modular communication API
+- IPC: Inter-process communication
+- sock: RIOT's primary API for network applications
+- POSIX: IEEE-standardized interface for operating systems compatibility
+- socket: POSIX' primary API for network applications
+
+## 1. Introduction
+At the moment of writing this document there are three APIs of note to have a
+node communicate over the network. From lowest level to highest these are:
+
+- GNRC's netapi
+- sock APIs
+- POSIX socket API
+
```

Comment:
I would list netdev in the above list with the others, for clarity. The following sentence would just be rephrased as "The netdev interface, meant to communicate with a network device driver, is not covered in detail by this document".